api/docs/download.dox

```diff
@@ -50,29 +50,29 @@ The source code is available:
 
 For the very latest changes since the last official release, you can download \ref page_weekly_builds.
 
-The [10.0.0 release](https://github.com/DynamoRIO/dynamorio/releases/tag/release_10.0.0):
+The [11.0.0 release](https://github.com/DynamoRIO/dynamorio/releases/tag/release_11.0.0):
```

Comment:
nit: Also change https://github.com/DynamoRIO/dynamorio/blob/master/api/docs/new_release.dox#L93 to 11.0.0? (Or explicitly say it's just an example?).

Related to this, I noticed that we have both `11.0` and `11.0.0` tags, and they point to the same commit. I think we only want to keep `11.0.0` (following the other tags).

Reply:
<img width="338" alt="Screenshot 2024-11-05 at 10 04 57 AM" src="https://github.com/user-attachments/assets/c3f36318-3ccb-42a4-9cc4-093b9bb70d7d">